### PR TITLE
chore: bump golang to 1.22

### PR DESCRIPTION
**Pull request by Xunzhuo**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Bump go version

### CODE REVIEW

File: tools/src/helm-docs/go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/envoyproxy/gateway/tools/src/helm-docs
 
-go 1.21
+go 1.22
 
 require github.com/norwoodj/helm-docs v1.12.0
```
Comment (zirain) [inline tools/src/helm-docs/go.mod:5]: maybe need bump to 1.13

**Comment by Xunzhuo** [comment]
@zirain crd-ref-docs cannot work in 1.22 ..

**Comment by Xunzhuo** [comment]
https://github.com/elastic/crd-ref-docs/pull/63

**Comment by zirain** [comment]
> [elastic/crd-ref-docs#63](https://github.com/elastic/crd-ref-docs/pull/63)

it's merged

**Review by zirain — APPROVED** [review]

**Pull request merged**